share/smack/frontend.py

```diff
@@ -0,0 +1,277 @@
+import os
+import sys
+from utils import temporary_file, try_command
+from svcomp.utils import svcomp_frontend
+
+def languages():
```

Comment:
I actually like this approach, and I think we should do it. However, in light of a few things I think we should kick this can down the road, and not include it in this PR.

First, I think you've described a good way to generalize the frontend code. But it will probably be good to also generalize the file structure of /share/smack/include and /share/smack/lib/, and we haven't thought a lot about this yet. Certainly it wouldn't be too hard, but this PR is starting to drag on.

Second, at 276 LOC, frontends.py hasn't fully achieved "giant monolith" status yet, although I imagine it will not take very long.

Third, we have an entire frontend, Rust, that's stuck in a branch and hasn't merged yet. I think waiting until that's in would make that merge easier, and would make the following changes more informed (what if rust has special considerations?).

So, to recap, I think we should start by getting these changes in, then expand and test the functionality for a short time (in particular, until rust is merged in, and we've worked on C++ issues, e.g. #361). Then we should make the changes you're talking about.

Reply:
Fine by me 👍